### DISCUSSION TITLE
What would be the best way to provide the templates?

### DISCUSSION THREAD

**Discussion by bentshrimp** [AgainIoT/Open-Set-Go#33] [Feature Suggestion / Share Ideas]
Open-Set-Go provides not only the project itself, but also presets for popular issue template, PR template, README.md, and CONTRIBUTING.md. I have discussed with @ymw0407 about if providing only document presets like now would be helpful for users. The conclusion was "no".

### We decided to provide categories for issue template presets, and PR template. And we agreed that it would be more helpful generating README.md and CONTRIBUTING.md for user's repository with project information than providing these documents included by existing repositories.
Check out the links below.
[https://readme.so/](https://readme.so/)
[https://www.genreadme.cloud/](https://www.genreadme.cloud/)

**Comment by ymw0407** [comment]
@AgainIoT/development-team
I think it would be good for maintainers to share their opinions on the discussion.
maintainer들은 해당 discussion에 대해서 의견을 나눠주면 좋을 것 같습니다.

**Comment by ymw0407** [comment]
## Contributing.md
The way we currently offer templates for Contributing.md is a bit unusual.
When I was planning for that part, I thought it could be generalized to a few specific types. However, when we proceeded with about 7 generalizations, it was not meaningless, but it was difficult to use because it was not easy to use and there were dependencies between types. That's why we're thinking about improving the way we currently provide templates to provide contributing.md in github/blog, so that we can provide the template itself and create it for individuals like readme.so with other options.
현재 Contributing.md에 대한 template을 제공하는 방식은 좀 특이합니다.
제가 해당 부분에 대해서 기획을 진행하였을 때, 특정 몇가지의 유형으로 일반화할 수 있다고 생각했습니다. 하지만, 일반화 7가지 정도로 진행을 해보았을때 무의미하지는 않았지만 활용하기 쉽지 않고 유형간에 종속이 존재했기 때문에 활용하기 어렵다는 판단을 하였습니다. 그렇기 때문에, 현재 template을 제공하는 방식을 개선하여 github/blog의 contributing.md를 제공한다고 하였을때, template 그 자체를 제공하고 다른 선택지로 readme.so와 같이 개인의 것에 맞게 제작할 수 있도록 하면 어떨까라는 생각을 하고 있습니다.

Or another way to personalize and provide templates of famous open sources. This, of course, makes it difficult to provide many templates. However, I think it will make the template easier to use.
혹은 다른 방법으로 유명 오픈소스들의 template을 개인화하여 제공하는 것입니다. 물론 이런 방법은 많은 template을 제공하기 어렵게 합니다. 하지만, 보다 template을 사용하기 쉽게 할 수 있을 것이라고 생각됩니다.

## ETC
I think the same goes for README and other templates...
README와 다른 template들도 마찬가지로 생각됩니다...

**Comment by sinji2102** [comment]
### Contributing.md
---
Today, Contributing.md categorized it into seven categories, but it was difficult to express all of the common Contributin.md . Therefore, I think it is better to provide a commonly used template such as [readme.so](https://readme.so/) as the next best option. At the same time, if you can offer multiple options within the same type, such as the **style** of **Contributing.md** within **Open-Set-Go**, I would like to recommend that you offer them like the **Github Profile** of **readme.so** .
현재 Contributing.md의 경우 7가지로 유형화하였으나, 이는 일반적인 Contributin.md를 모두 표현하기에는 어려움이 있었습니다. 때문에 차선책으로 [readme.so](https://readme.so/)와 같이 일반적으로 사용되는 template를 제공하는 것이 좋다고 생각합니다. 동시에 **Open-Set-Go** 내 **Contributing.md**의 **style**과 같이 같은 유형 내에 여러 선택지를 제공할 수 있는 경우 **readme.so**의 **Github Profile**과 같이 제공하는 방식을 추천하고 싶습니다.


### README
---
Currently, README.md offers a well-known open source template, but since it's a difficult structure for individuals to customize, it's available in sections like [readme.so](https://readme.so/) , but I think it's a great way to increase options within sections.
현재 README.md의 경우 유명 오픈소스들의 template을 제공하고 있습니다. 그러나 이는 개인이 사용자화하기 어려운 구조라고 판단되어 [readme.so](https://readme.so/)와 같이 섹션을 나눠서 제공하되, 섹션 내에 선택지를 늘리는 방식이 좋다고 생각합니다.

**Reply by bentshrimp** [reply]
I strongly agree with you. Categorizing README.md and CONTRIBUTING.md seems meaningless.
